base/special/hyperbolic.jl

```diff
@@ -0,0 +1,251 @@
+# ====================================================
+# Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+#
+# Developed at SunSoft, a Sun Microsystems, Inc. business.
+# Permission to use, copy, modify, and distribute this
+# software is freely granted, provided that this notice
+# is preserved.
+# ====================================================
+
+# Hyperbolic functions
+# sinh methods
+SINH_SMALL_X(::Type{Float64}) = 2.0^-28
+H_MEDIUM_X(::Type{Float64}) = 22.0
+
+SINH_SMALL_X(::Type{Float32}) = 2f-12
+H_MEDIUM_X(::Type{Float32}) = 9f0
+
+H_LARGE_X(::Type{Float64}) = 709.7822265633563 # nextfloat(709.7822265633562)
+H_OVERFLOW_X(::Type{Float64}) = 710.475860073944 # nextfloat(710.4758600739439)
+
+H_LARGE_X(::Type{Float32}) = 88.72283f0
+H_OVERFLOW_X(::Type{Float32}) = 89.415985f0
+function sinh(x::T) where T <: Union{Float32, Float64}
```

Comment:
Wouldn't it be cleaner to do `sinh(x::Union{Float32, Float64})` for functions like this?

Reply:
yes I use these to pick the branching cutoffs. Do you think it would be cleaner to write `sinh(x::Union{Float32, Float64}`  and then just define `H_LARGE_X(::Float32) = 88.72283f0`? on the actual `x` instead of the type of `x`? Should compile down to the same thing, right?

Reply:
I think it's fine as-is.